scripts/restart_mcp_server.sh

```diff
@@ -6,19 +6,67 @@ CONTAINER_NAME=${CONTAINER_NAME:-stt-mcp-server-linux}
 TMUX_SESSION=${TMUX_SESSION:-claude}
 TMUX_TMPDIR=${TMUX_TMPDIR:-"$HOME"/.tmux}
 
-docker stop "$CONTAINER_NAME" || true
-docker rm "$CONTAINER_NAME" || true
+# Check if Docker is running
+if ! docker info >/dev/null 2>&1; then
+    echo "Error: Docker is not running or not accessible" >&2
+    exit 1
+fi
+
+# Stop and remove container if it exists
+if docker ps -a --format '{{.Names}}' | grep -q "^${CONTAINER_NAME}$"; then
+    echo "Stopping and removing existing container: $CONTAINER_NAME"
+    docker stop "$CONTAINER_NAME" || true
+    docker rm "$CONTAINER_NAME" || true
+else
+    echo "Container $CONTAINER_NAME does not exist, proceeding with creation"
+fi
+
+# Create required directories
+mkdir -p "$HOME/.whisper"
+mkdir -p "$TMUX_TMPDIR"
+
+# Get input group ID for device access
+INPUT_GID=$(getent group input | cut -d: -f3)
+
+# Build Docker command using array for better handling of arguments
```

Comment:
Please remove the vacuous comments. I believe no comments are necessary in this file, except for this one, since bash arrays may be unfamiliar to some users.

However, the word "better" is not necessay here, as it does not add new information (like better than what).